.github/workflows/ci.yml

```diff
@@ -0,0 +1,199 @@
+name: CI
+run-name: ${{ inputs.model }} CI
+# NOTE: This workflow requires:
+# permissions.pull-requests:write
+# permissions.contents:write
+# secrets:inherit with an appropriate GitHub Environment for deployment in the caller
+on:
+  workflow_call:
+    inputs:
+      model:
+        type: string
+        required: true
+        description: The model that is being tested and deployed
+      root-sbd:
+        type: string
+        required: false
+        default: ${{ inputs.model }}
+        description: |
+          The name of the root Spack Bundle Definition, if it is different from the model name.
+          This is often a package named similarly in ACCESS-NRI/spack-packages.
+  # Callers usually have the trigger:
+  # pull_request:
+  #   branches:
+  #     - main
+  #     - backport/*.*
+  #   paths:
+  #     - config/**
+  #     - spack.yaml
+
+env:
+  SPACK_YAML_MODEL_YQ: .spack.specs[0]
+jobs:
+  validate-json:
+    name: Validate JSON
+    uses: access-nri/actions/.github/workflows/validate-json.yml@main
+    with:
+      src: "config"
+
+  check-config:
+    name: Check Config Fields
+    needs:
+      - validate-json
+    runs-on: ubuntu-latest
+    outputs:
+      spack-packages-version: ${{ steps.spack-packages.outputs.version }}
+      spack-config-version:  ${{ steps.spack-config.outputs.version }}
+    steps:
+      - name: Validate spack-packages version
+        id: spack-packages
+        uses: access-nri/build-cd/.github/actions/validate-repo-version@main
+        with:
+          repo-to-check: spack-packages
```

Comment:
Just want to check this will work if the workflow is called from outside the ACCESS-NRI organisation.

Reply:
Currently, no it won't. This is because we are comparing the version specified in the `versions.json` like so: https://github.com/ACCESS-NRI/ACCESS-OM2/blob/main/config/versions.json#L3
Which doesn't account for a repo owner (we assume it to be ACCESS-NRI).

Would we want it to be used more widely? We would have to update each models `versions.json` to be Something like:
```json
"ACCESS-NRI/spack-packages": "2023.02.11"
```

Reply:
I was wondering how this might affect the CABLE-LSM/CABLE model if the context it was calling it from couldn't find `spack-packages`. I guess though that the `build-cd` action is being called from the ACCESS-NRI org so it's not an issue?

Reply:
It should be able to find it - it's a public repository in our org. If that's what you meant?